ci/sdk_elixir.go

```diff
@@ -0,0 +1,156 @@
+package main
+
+import (
+	"context"
+	"dagger/internal/dagger"
+	"fmt"
+	"regexp"
+	"strings"
+
+	"golang.org/x/sync/errgroup"
+)
+
+const (
+	elixirSDKPath            = "sdk/elixir"
+	elixirSDKGeneratedPath   = elixirSDKPath + "/lib/dagger/gen"
+	elixirSDKVersionFilePath = elixirSDKPath + "/lib/dagger/core/engine_conn.ex"
+)
+
+// https://hub.docker.com/r/hexpm/elixir/tags?page=1&name=debian-buster
+var elixirVersions = []string{"1.16.2", "1.15.7", "1.14.5"}
+
+const (
+	otpVersion    = "26.2.3"
+	debianVersion = "20240130"
+)
+
+type ElixirSDK struct {
+	Dagger *Dagger // +private
+}
+
+// Lint lints the Elixir SDK
+func (t ElixirSDK) Lint(ctx context.Context) error {
+	ctr, err := t.Dagger.installDagger(ctx, t.elixirBase(elixirVersions[1]), "sdk-elixir-lint")
+	if err != nil {
+		return err
+	}
+
+	_, err = ctr.
+		WithExec([]string{"mix", "lint"}).
+		Sync(ctx)
+	return err
+}
+
+// Test tests the Elixir SDK
+func (t ElixirSDK) Test(ctx context.Context) error {
+	ctrs := []*dagger.Container{}
+	for _, elixirVersion := range elixirVersions {
+		ctrs = append(ctrs, t.elixirBase(elixirVersion))
+	}
+	ctrs, err := t.Dagger.installDaggers(ctx, ctrs, "sdk-elixir-test")
+	if err != nil {
+		return err
+	}
+
+	eg, ctx := errgroup.WithContext(ctx)
+	for _, ctr := range ctrs {
+		ctr := ctr
+		eg.Go(func() error {
+			_, err := ctr.
+				WithExec([]string{"mix", "test"}).
+				Sync(ctx)
+			return err
+		})
+	}
+	return eg.Wait()
+}
+
+// Generate re-generates the Elixir SDK API
+func (t ElixirSDK) Generate(ctx context.Context) (*Directory, error) {
+	ctr, err := t.Dagger.installDagger(ctx, t.elixirBase(elixirVersions[1]), "sdk-elixir-generate")
+	if err != nil {
+		return nil, err
+	}
+
+	gen := ctr.
+		WithExec([]string{"mix", "run", "scripts/fetch_introspection.exs"}).
```

Comment:
@helderco suggested me to uses `github.com/helderco/daggerverse/codegen` to fetch introspection. So that we can drop this script in favor of that function. 😸

Reply:
We cannot use this right now. The introspection that return from that function is different than the module because that function try to unify json to same format (both module and non-module). To use that function, we need to modify the Elixir codegen to handle json from that function and then use it. So my conclusion is postpone to the next PR since our goal is to port mage into the module.